src/app/util/mock/attribute-storage.cpp

```diff
@@ -0,0 +1,187 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *     This file contains the mock implementation for the generated attribute-storage.cpp
+ *     - It contains three endpoints, 0xFFFE, 0xFFFD, 0xFFFC
+ *     - It contains four clusters: 0xFFF1'0001 to 0xFFF1'0004
+ *     - All cluster has two global attribute (0x0000'FFFC, 0x0000'FFFD)
+ *     - Some clusters has some cluster-specific attributes, with 0xFFF1 prefix.
+ */
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app/util/mock/Constants.h>
+
+#include <app/ClusterInfo.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/EventManagement.h>
+#include <app/InteractionModelDelegate.h>
+#include <lib/core/CHIPCore.h>
+#include <lib/core/CHIPTLVDebug.hpp>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/DLLUtil.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+// Mock Functions
+typedef uint8_t EmberAfClusterMask;
+#define CLUSTER_MASK_SERVER (0x40)
```

Comment:
Can we not just include the right headers?

Also, these are not functions.

Reply:
Yes, this is inside app-common